app/jobs/dhis2/ethiopia_exporter_job.rb

```diff
@@ -1,23 +1,100 @@
-        htn_ltfu: PatientStates::Hypertension::LostToFollowUpPatientsQuery.new(region, period).call.count,
-        htn_dead: PatientStates::Hypertension::DeadPatientsQuery.new(region, period).call.count,
-        htn_cumulative_registrations: PatientStates::Hypertension::CumulativeRegistrationsQuery.new(region, period).call.count,
-        htn_monthly_registrations: PatientStates::Hypertension::MonthlyRegistrationsQuery.new(region, period).call.count,
-        htn_cumulative_assigned_adjusted: PatientStates::Hypertension::AdjustedAssignedPatientsQuery.new(region, period).call.count
+        htn_enrolled_under_care: {category_option_key: "dhis2_age_gender_category_elements", values: disaggregate_by_gender_age(under_care_patients, AGE_BUCKETS)},
+        htn_enrolled_under_care_treatment: {category_option_key: "dhis2_treatment_category_elements", values: segregate_and_format_treatment_data(under_care_patients)},
+        htn_by_enrollment_time: {category_option_key: "dhis2_enrollment_time_category_elements", values: segregate_and_format_enrollment_data(PatientStates::Hypertension::CumulativeRegistrationsQuery.new(region, period).call)},
+        htn_cohort_registered: {category_option_key: "dhis2_cohort_registered_category_elements", values: format_cohort_registered_data(cohort_registered_patients)},
+        htn_cohort_outcome: {category_option_key: "dhis2_cohort_category_elements", values: segregate_and_format_cohort_data(cohort_registered_patients)}
       }
     end
 
     def data_elements_map
       CountryConfig.dhis2_data_elements.fetch(:dhis2_data_elements)
     end
+
+    def category_element_map(category_element_key)
+      CountryConfig.dhis2_data_elements.fetch(category_element_key.to_sym)
+    end
+
+    def attribute_option_map_id
+      CountryConfig.dhis2_data_elements.fetch(:dhis2_attribute_option)
+    end
+
+    def format_facility_period_data(facility_data, facility_identifier, period)
+      formatted_facility_data = []
+      facility_data.each do |data_element, options|
+        attribute_option_id = attribute_option_map_id
+        if options[:category_option_key]
+          category_element_map(options[:category_option_key]).each do |category_key, id|
+            formatted_facility_data << {
+              data_element: data_elements_map[data_element],
+              org_unit: facility_identifier.identifier,
+              category_option_combo: id,
+              period: reporting_period(period),
+              attribute_option_combo: attribute_option_id,
+              value: options[:values][category_key] || 0
+            }
+          end
+        else
+          formatted_facility_data << {
+            data_element: data_elements_map[data_element],
+            org_unit: facility_identifier.identifier,
+            period: reporting_period(period),
+            attribute_option_combo: attribute_option_id,
+            value: options[:value] || 0
+          }
+        end
```

Comment:
[suggestion, non-blocking] I looked at the caller again, and given that all our facility data from `facility_data_for_period` has a `category_option_key` this else branch here will never be triggered. Can we eliminate this and simplify the code in that case?